src/Microsoft.Extensions.WebSockets.Internal/WebSocketConnection.cs

```diff
@@ -249,8 +247,18 @@ private void WriteMaskingKey(Span<byte> buffer)
             _maskingKeyBuffer.CopyTo(buffer);
         }
 
+        /// <summary>
+        /// Terminates the socket abruptly.
+        /// </summary>
+        public void Abort()
+        {
+            _terminateReceiveCts.Cancel();
```

Comment:
If you aren't currently reading, the next read will be completed synchronously with a cancellation.